### PR TITLE
updated community rules to display the actual rules

### LINKED ISSUE #18978: Implement context action "View community rules"
As a user,
I want to be able to see the community rules,
So I know how to behave in the community.

### Description

Currently we show some canned text for community rules, the actual rules should be displayed.

https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=15927-431360&mode=design&t=j9xL4xn1wE0KKjY0-4

https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=1388-104275&mode=design&t=nBBw5iB9Gl8L6cGt-0

### PR DESCRIPTION
**Pull request by jo-mut**
fixes [#18978](https://github.com/status-im/status-mobile/issues/18978)

### Summary

This allows the user to view the actual community rules set when the community was created

This pr. would appreciate a design review

### CODE REVIEW

**Review by ajayesivan — COMMENTED** [review]
Scroll issue: Unable to scroll to the bottom of the community rules when the rules are longer than the screen height.

https://github.com/status-im/status-mobile/assets/19279756/b3432816-b293-4d07-8d34-4ad0bae895f7

**Comment by jo-mut** [comment]
@ajayesivan the scroll issue on the bottom sheet is resolved

https://github.com/status-im/status-mobile/assets/28704507/4f42b10c-dbe7-44ff-9425-3f54a94af945

File: src/status_im/contexts/communities/actions/generic_menu/view.cljs [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
+        max-bottom-sheet-height                     (- window-height 44)
+        ;; This is the default height of the bottom sheet with zero nested views
+        empty-bottom-sheet-height                   62
+        total-bottom-sheet-height                   (+ scroll-view-height empty-bottom-sheet-height)]
+    [gesture/scroll-view
+     {:on-layout                       (fn [event]
+                                         (set-scroll-view-height (oops/oget
+                                                                  event
+                                                                  "nativeEvent.layout.height")))
+      :scroll-enabled                  (when (>= total-bottom-sheet-height max-bottom-sheet-height) true)
```
Comment (ajayesivan) [inline src/status_im/contexts/communities/actions/generic_menu/view.cljs]: We don't need to use `when` here. The output of `(>= total-bottom-sheet-height max-bottom-sheet-height)` will be either `true `or `false`.

**Review by ajayesivan — APPROVED** [review]
Nice work 🎉

**Comment by mariia-skrypnyk** [comment]
Hi @jo-mut ! 
Thanks for your PR!

Here are some issues I've found.
Please let me know if you need extra details or you have comments regarding current implementation.

**ISSUE 1: no standard text in community rules of all types for Android** 

**Steps:**

1. Login to app
2. Create communities of all types
3. Make a long tap on your community on Communities screen
4. Choose Communities Rules

**Now**
photo_2024-03-22_11-19-37

**Design**
Screenshot 2024-03-22 at 11 24 51

**ISSUE 2: Margins looks bigger as designed and seems like we use the wrong font size**

**Steps:**

1. Login to app
2. Create community
3. Make a long tap on your community on Communities screen
4. Choose Communities Rules

**Now vs Design**
Screenshot 2024-03-22 at 10 10 06

**ISSUE 3: No community rules for community created from the desktop for iOS**

1. Login to app
2. Create community on he desktop and write some rules for it
3. As a mobile user join the desktop community
4. Make a long tap on your community on Communities screen
5. Choose Communities Rules

**Mobile rules**
photo_2024-03-22_11-30-23

**Desktop rules**
photo_2024-03-22_11-30-29

**Comment by jo-mut** [comment]
@mariia-skrypnyk I have gone through your feedback and I do thing there are somethings that deserve some clarification.

**ISSUE 1:**
I was under the impression that the Standard rules are no longer needed since the user is required to submit `a community intro message and community rules` at the creation of the community. Which means at no point will the standard rules be displayed to the user since. Issue 1 is only reproducible in the case of where the community was created from mobile and for this case I have updated the code so that the standard rules would be displayed

**ISSUE 2:**
I noticed there are two different designs for the same screen and I relied on this one  https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=13995%3A838366&mode=design&t=HBwmC3ube6Ayt9Sv-1 which is different from the one you used for testing and reproduction of issue 2. The difference in the `bottom-margin` in these screens differs because of the `bottom-inset` which is determined by the `os`. Going by the design I worked with, I think there is not further action to takes as the margins match. I stand to be corrected on this should one of the design be not correct

Screenshot 2024-03-24 at 21 53 52

**ISSUE 3:**
Referrencing ISSUE 1: at the moment when the user is creating a community from desktop, the user is required to input a community message and rules. It's a must fill input-filed on desktop. The standard rules would therefore not be displayed on mobile unless the community was created on mobile in the updated code

**Mobile community** vs **Desktop community**

**Comment by jo-mut** [comment]
@mariia-skrypnyk ISSUE 2 its more of a design bug, i will update the code to reflect your feedback

**Comment by mariia-skrypnyk** [comment]
Hi @jo-mut !

**ISSUE 1:** I really understand your vision regarding basic hardcoded rules. 
But I think if we do have it added for iOS let it be also done for Android. 
Yes, the real user will see only rules added by community author. 
And we do have a strong purpose to use community creation from mobile in testing purpose. 
So, if it will be an easy fix - let this hardcoded text be present on both platforms. 
Does it works for you?

**ISSUE 2:** let's leave it until the design review as if we have different designs it needs to be clarified.

**Comment by jo-mut** [comment]
> Hi @jo-mut !
> 
> **ISSUE 1:** I really understand your vision regarding basic hardcoded rules. But I think if we do have it added for iOS let it be also done for Android. Yes, the real user will see only rules added by community author. And we do have a strong purpose to use community creation from mobile in testing purpose. So, if it will be an easy fix - let this hardcoded text be present on both platforms. Does it works for you?

yeah I already update the code to display the rules. thanks Maria

**Comment by jo-mut** [comment]
@mariia-skrypnyk  actually on ISSUE 2: @mariocnovoa has already reviewed and updated the figma designs to resolve the bug so I will update the code to reflect address the bug too

**Comment by mariia-skrypnyk** [comment]
@jo-mut not sure I understood you with **ISSUE 3**:

iOS User doesn't see community rules that were added from the desktop, he sees hardcoded text.
But he should see these rules.

Let me know if you have another vision of such behaviour.

User A is a desktop user
User  A creates community with rules
User B joins this community
Expected: User B should see rules that were added by User A 
Actual: User B sees hardcoded text

**Comment by jo-mut** [comment]
> @jo-mut not sure I understood you with **ISSUE 3**:
> 
> iOS User doesn't see community rules that were added from the desktop, he sees hardcoded text. But he should see these rules.
> 
> Let me know if you have another vision of such behaviour.
> 
> User A is a desktop user User A creates community with rules User B joins this community Expected: User B should see rules that were added by User A Actual: User B sees hardcoded text

Interesting. I am testing this on iOS and as User B I can see the rules on my device created by user A on desktop as you can see my screenshots are iOS . I wonder why you dont see the rules on User B iOS device

**Comment by mariia-skrypnyk** [comment]
@jo-mut thanks for checking ISSUE 3.
As new builds are available I'll recheck them and let you know!

**Comment by mariia-skrypnyk** [comment]
@jo-mut I've checked text and now ISSUE 3 is closed.

But I see issue with scroll on iOS.
Device: iPhone 13 17.3.1

ISSUE 4: no scroll for rules on iOS

Actual result:
Screenshot 2024-03-25 at 16 19 22

Here is how rules are written in desktop community

https://github.com/status-im/status-mobile/assets/147824399/93d0d594-4c78-4185-b3b6-ce4cf86e4272

**Comment by mariia-skrypnyk** [comment]
@jo-mut updated video above, it was not loaded.

**Comment by mariia-skrypnyk** [comment]
Hi @jo-mut can I test your fixes?

**Comment by jo-mut** [comment]
oooh sorry, yes please @mariia-skrypnyk

**Comment by jo-mut** [comment]
@mariia-skrypnyk ass for ISSUE 2: on the bottom margin, I looked at most of the bottom sheet implementation vs design and most of the implementations if not all have an extra margin on iOS, I saw we skip this spacing issues for this pr and it can be sorted out all together in a separate issue.

**Comment by mariia-skrypnyk** [comment]
Thanks @jo-mut !

Very appreciate your work.
I've checked Issues 3 and 4:

✅ Issue 3
Issue 4 : text at the bottom is very near to the end of the screen, I have iPhone 13 and here how it is there
Screenshot 2024-04-02 at 13 03 11
photo_2024-04-02_13-04-23

Can you add more space to the bottom when scrolling?

**Comment by jo-mut** [comment]
@mariia-skrypnyk should now be fine

https://github.com/status-im/status-mobile/assets/28704507/40090801-c830-4bf5-9564-5f47bcd200a5

**Comment by mariia-skrypnyk** [comment]
@jo-mut no new builds from today, will they appear soon?

**Comment by jo-mut** [comment]
@mariia-skrypnyk i am not sure i understand your question, could you maybe rephrase it

**Comment by mariia-skrypnyk** [comment]
> @mariia-skrypnyk should now be fine

Sure)

@jo-mut you wrote that you fixed marging.
I thought this fix will be in a fresh build from 02/04/2024 but the last one is from 01/04/2024
Should I use the last build?

**Comment by jo-mut** [comment]
@mariia-skrypnyk oooh makes sense, i posted the screen-recording and didn't push, i have updated

**Comment by mariia-skrypnyk** [comment]
Thanks @jo-mut !
iOS ✅ from my side
Android is near to the boarder but as for me we need @Francesca-G look and approve.
Now Android scroll looks like this:

https://github.com/status-im/status-mobile/assets/147824399/c610c713-7cd1-44b9-9245-df0c84230a35

**Review by Francesca-G — APPROVED** [review]
Bottom spacing doesn't match design:

Screenshot 2024-04-02 alle 17 28 59

>
besides that it looks good to me👍

**Comment by jo-mut** [comment]
> Bottom spacing doesn't match design:
> 
> Screenshot 2024-04-02 alle 17 28 59
> > 
> 
> besides that it looks good to me👍

I will create an issue for this as it affects most of the bottom sheets
